### PR TITLE
nullptr initialisation of FormatArg pointers, runtime debug mode checks

### PR DESCRIPTION
**Pull request by nigels-com**
Proposed fix for Issue #42 

Initialise FormatArg pointers in default constructor, runtime assertions.

### CODE REVIEW

**Comment by c42f** [comment]
Thanks, looks fine to me.

File: tinyformat.h [hunk 1]
```diff
@@ -846,6 +854,7 @@ class FormatList
                             const FormatList& list);
 
     private:
+        FormatList();
```
Comment (krlmlr) [inline tinyformat.h]: Why is this necessary?

Comment (c42f) [reply tinyformat.h]: Good question actually.  @nigels-com why did you add this?

Comment (nigels-com) [reply tinyformat.h]: A FormatList default constructor looks like another place that ought to be doing initialisation.

Comment (c42f) [reply tinyformat.h]: The default constructor is deleted because a nontrivial constructor was declared, so this should be unncessary.  I'll remove it.

Comment (nigels-com) [reply tinyformat.h]: Better to be explicit, IMHO.
